### ISSUE TITLE
Rule proposal: disallow string interpolation for className

### ISSUE THREAD

**Issue by lencioni** [yannickcr/eslint-plugin-react#460]
Using string interpolation in classNames can make it difficult to determine whether some CSS is actually used in your project. To make this easier, some folks might want to enforce preventing string interpolation in classNames.

Bad:

```jsx
const MyComponent = function({ someProp }) {
  const classNames = `my-component my-component-${someProp}`;

  return (
    <div className={classNames}>
      Hi mom
    </div>
  )
};
```

Bad:

```jsx
const MyComponent = function({ someProp }) {
  const classNames = 'my-component my-component-' + someProp;

  return (
    <div className={classNames}>
      Hi mom
    </div>
  )
};
```

Bad:

```jsx
const MyComponent = function({ someProp }) {
  const classNames = ['my-component'];
  classNames.push('my-component-' + someProp);

  return (
    <div className={classNames.join(' ')}>
      Hi mom
    </div>
  )
};
```

Good:

```jsx
const MyComponent = function({ someProp }) {
  const classNames = ['my-component'];
  if (someProp === 'some-value') {
    classNames.push('my-component-some-value');
  } else if (someProp === 'another-value') {
    classNames.push('my-component-another-value');
  }

  return (
    <div className={classNames.join(' ')}>
      Hi mom
    </div>
  )
};
```

Of course, the [classnames package](https://github.com/JedWatson/classnames) can be useful here. 

I don't know yet, but this may be useful for other props, so perhaps it could be configured for a list of props.

**Comment by Katerina198b** [comment]
I have the same proposal)
Maybe there are new rules for this problem already?
